taf/updater/updater.py

```diff
@@ -186,6 +186,21 @@ class RepositoryConfig:
         default=False,
         metadata={"docs": "Whether update fails if a warning is raised. Optional."},
     )
+    # JMC: Addition of --no-deps option to disallow updating of the last validated commit if any part of validation is skipped
+    no_deps: bool = field(
+        default=False,
+        metadata={"docs": "Specifies whether or not to update dependencies. Optional."}
+    )
+    # JMC: Addition of --no-targets option to allow user to skip target repos when validating the authentication repository.
+    no_targets: bool = field(
+        default=False,
+        metadata={"docs": "Flag to skip target repositiory validation and validate only authentication repos. Optional."}
+    )
+    # JMC: Addition of --no-upstream option to allow user to opt out of comparing with the remote repository and be added to clone and update
+    no_upstream: bool = field(
+        default=False,
```

Comment:
This feature was not very well described in the issue. What this is supposed to do is only opt out of comparing target repositories with upstream, if case there were no changes pushed to the authentication repository. If there were no changes pushed to the auth repo, then we are sure that there are no valid changes pushed to the target repositories and we are sure that nothing will be pulled down. However, by comparing the local target repositories with the remote repositories, we can detect errors. Only some users need to do this. This should be set too `True` by default. When we only want to validate local repositories, we can run `taf repo validate`